src/channel.c

```diff
@@ -581,3 +592,41 @@ IMPLEMENT_MOCKABLE_FUNCTION(, CHANNEL_RESULT, channel_push, THANDLE(CHANNEL), ch
 
     return result;
 }
+
+uint32_t channel_get_count_of_items_in_channel(THANDLE(CHANNEL) channel)
+{
+    uint32_t result;
+    if (channel == NULL)
+    {
+        LogError("Invalid arguments: THANDLE(CHANNEL) channel=%p", channel);
+        /*Codes_SRS_CHANNEL_18_008: [ If channel is NULL, channel_get_count_of_items_in_channel shall return UINT32_MAX. ]*/
+        result = UINT32_MAX;
+    }
+    else
+    {
+        CHANNEL* channel_ptr = THANDLE_GET_T(CHANNEL)(channel);
+        /*Codes_SRS_CHANNEL_18_009: [ channel_get_count_of_items_in_channel shall return the count of items in the channel. ]*/
+        result = interlocked_add(&channel_ptr->count_of_items_in_channel, 0);
+    }
+    return result;
+}
+
+int channel_seal_channel(THANDLE(CHANNEL) channel)
+{
+    int result;
+    if (channel == NULL)
+    {
+        /*Codes_SRS_CHANNEL_18_010: [ If channel is NULL, channel_seal_channel shall fail and return a non-zero value. ]*/
+        LogError("Invalid argument: THANDLE(CHANNEL) channel=%p", channel);
+        result = MU_FAILURE;
+    }
+    else
+    {
+        CHANNEL* channel_ptr = THANDLE_GET_T(CHANNEL)(channel);
+        /*Codes_SRS_CHANNEL_18_011: [ channel_seal_channel shall set the channel as sealed. ]*/
+        (void)interlocked_exchange(&channel_ptr->is_sealed, 1);
```

Comment:
>(void)interlocked_exchange(&channel_ptr->is_sealed, 1); [](http://example.com/codeflow?start=8&length=55)

What if the channel is not open? Is seal safe?

Reply:
new iteration only allows seal if the channel is open.